cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -74,8 +75,8 @@ type ClusterState struct {
 
 // GetAvailableUpgrades fetches all versions from the specified VersionGetter and computes which
 // kinds of upgrades can be performed
-func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesAllowed, rcUpgradesAllowed bool, etcdClient etcdutil.ClusterInterrogator, dnsType kubeadmapi.DNSAddOnType, client clientset.Interface) ([]Upgrade, error) {
-	fmt.Println("[upgrade] Fetching available versions to upgrade to")
+func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesAllowed, rcUpgradesAllowed bool, etcdClient etcdutil.ClusterInterrogator, dnsType kubeadmapi.DNSAddOnType, client clientset.Interface, printer output.Printer) ([]Upgrade, error) {
```

Comment:
[2] I think we can remove printer here and move the two printf out of this call

Reply:
will do